modules/xdcr-reference/pages/xdcr-filtering-data-type-conversion.adoc

```diff
@@ -0,0 +1,479 @@
+| Comparison is performed along the following lines: if the length of `Array_1` is greater than the length of `Array_2`, then `Array_1 > Array_2` returns `true`.
+If the arrays are of equal length, corresponding objects are compared in sequence, until one is found to be the greater, and `true` is returned &#8212; _i.e._ `Array_1[object_1] > Array_2[object_1]`, `Array_1[object_2] > Array_2[object_2]`, and so forth.
+
+| 26
+| `Object`
+| `Object`
+| image:xdcr-reference:dash-black-inline.png[] No conversion need be performed.
+| Comparison is performed along the following lines: if the length of `Object_1` is greater than the length of `Object_2`, then `Object_1 > Object_2` returns `true`.
+If the objects are of equal length, corresponding bytes of the objects' data are compared in sequence, until one is found to be the greater, and `true` is returned &#8212; _i.e._ `Object_1[byte_1] > Object_2[byte_1]`, `Object_1[byte_2] > Object_2[byte_2]`, and so forth.
+
+|===
+
+
+[#implicit-conversion-modes]
+== Implicit Conversion Modes
+
+Each filter expression requires implicit conversion to be applied to one of the following combinations:
+
+* _A Constant and a Variable_.
+The filter expression contains a user-specified constant and a variable, which are to be compared.
+Couchbase Server determines the data type of the constant, and attempts to apply this data type to the value of the variable.
+
+* _Two Variables_.
+The filter expression specifies two variables, which are to be compared.
+Couchbase Server determines the data type of the variables from the corresponding values in the JSON document to which filtering is currently being applied.
+
+These modes are described in the subsections below.
+
+[#implicit-conversion-of-constant-and-variable]
+=== Implicit Conversion of Constant and Variable.
+
+When a constant and a variable are to be compared, Couchbase Server determines the data type of the constant, and attempts to apply this also to the value of the variable.
+
+[#data-type-conversion-of-user-specified-constants]
+==== Data-Type Conversion of User-Specified Constants
+
+When the user explicitly enters a _constant_ into a filter expression, the data type for the constant is evaluated by Couchbase Server, as part of its process for _tokenizing_ the expression (that is, parsing the expression into identifiable lexical components).
+
+The correspondences between token formats and duly assigned data types is described in the following table.
+
+[cols="2,2,2"]
+|===
+| *Token Format* | *Assigned Data Type* | *Example*
+
+| Any character-sequence enclosed either by double quotes (`""`) or by single quotes (`''`).
+| `String`
+| `variable == "a string"`
+
+`Variable == 'another string'`
+
+| Any numeric values without precision delimiter or mantissa.
+| `Int`
+| `variable > 1234`
+
+`variable < -2345`
+
+| Any number representing a valid `golang` float, optionally with precision delimiter and/or mantissa.
+| `Float`
+| `variable >= 1.2343+25`
```

Comment:
> 1.2343+25

Should be 1.2343**e**+25
https://stackoverflow.com/questions/19554972/json-standard-floating-point-numbers/19554986#19554986

Reply:
Ok. Done.